src/features/fileIssue.ts

```diff
@@ -0,0 +1,112 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { vscode } from "../vscodeAdapter";
+import { Extension } from "../vscodeAdapter";
+import { CSharpExtensionId } from "../constants/CSharpExtensionId";
+import { EventStream } from "../EventStream";
+import { ReportIssue } from "../omnisharp/loggingEvents";
+
+const issuesUrl = "https://github.com/OmniSharp/omnisharp-vscode/issues/new";
+
+export default async function fileIssue(vscode: vscode, eventStream: EventStream, execChildProcess: (command: string, workingDirectory?: string) => Promise<string>, isValidPlatformForMono: boolean) {
+    const dotnetInfo = await getDotnetInfo(execChildProcess);
+    const monoInfo = await getMonoIfPlatformValid(execChildProcess, isValidPlatformForMono);
+    let extensions = getInstalledExtensions(vscode);
+    let csharpExtVersion = getCsharpExtensionVersion(vscode);
+
+    const body = encodeURIComponent(`## Issue Description ##
+## Steps to Reproduce ##
+
+## Expected Behaviour ##
+
+## Actual Behaviour ##
+
+## Logs ##
+
+### OmniSharp log ###
+<details>Post the output from Output-->OmniSharp log here</details>
+
+### C# log ###
+<details>Post the output from Output-->C# here</details>
+
+## Environment information ##
+
+**VSCode version**: ${vscode.version}
+**C# Extension**: ${csharpExtVersion}
+
+${monoInfo}
+<details><summary>Dotnet Information</summary>
+${dotnetInfo}</details>
+<details><summary>Visual Studio Code Extensions</summary>
+${generateExtensionTable(extensions)}
+</details>
+`);
+
+    eventStream.post(new ReportIssue(issuesUrl, body));
+}
+
+function sortExtensions(a: Extension<any>, b: Extension<any>): number {
+
+    if (a.packageJSON.name.toLowerCase() < b.packageJSON.name.toLowerCase()) {
+        return -1;
+    }
+    if (a.packageJSON.name.toLowerCase() > b.packageJSON.name.toLowerCase()) {
+        return 1;
+    }
+    return 0;
+}
+
+function generateExtensionTable(extensions: Extension<any>[]) {
+    if (extensions.length <= 0) {
+        return "none";
+    }
+
+    const tableHeader = `|Extension|Author|Version|\n|---|---|---|`;
+    const table = extensions.map((e) => `|${e.packageJSON.name}|${e.packageJSON.publisher}|${e.packageJSON.version}|`).join("\n");
+
+    const extensionTable = `
+${tableHeader}\n${table};
+`;
+
+    return extensionTable;
+}
+
+async function getMonoIfPlatformValid(execChildProcess: (command: string, workingDirectory?: string) => Promise<string>, isValidPlatformForMono: boolean): Promise<string> {
+    if (isValidPlatformForMono) {
+        let monoInfo: string;
+        try {
+            monoInfo = await getMonoVersion(execChildProcess);
+        }
+        catch (error) {
+            monoInfo = "A valid mono installation could not be found. Using the built-in mono"
```

Comment:
The extension has logic that determines the version of mono found and only using it if it's new enough. Can we try to follow the same logic here? We don't want to report mono versions that are old enough that we use the built in one instead.